src/ProjectService.Business/Commands/GetTaskCommand.cs

```diff
@@ -0,0 +1,176 @@
+            
+            Authorization(task, errors, out IGetDepartmentResponse department);
+
+            List<Guid> userIds = new ()
+            {
+                task.AuthorId,
+            };
+            
+            if (task.AssignedUser != null)
+            {
+                userIds.Add(task.AssignedUser.Id);
+            }
+
+            if (task.ParentTask != null)
+            {
+                userIds.Add(task.ParentTask.AuthorId);
+            }
+
+            var parentTaskAssignedTo = task.ParentTask?.AssignedTo.GetValueOrDefault();
+            if (parentTaskAssignedTo != null && parentTaskAssignedTo != Guid.Empty)
+            {
+                userIds.Add(parentTaskAssignedTo.Value);
+            } 
+            
+            var res = _usersDataRequestClient.GetResponse<IOperationResult<IGetUsersDataResponse>>(
+                IGetUsersDataRequest.CreateObj(userIds));
+
+            var usersDataResponse = res.Result.Message.Body.UsersData;
+            
+            List<TaskInfo> subtasksInfo = new ();
+            if (task.Subtasks != null)
+            {
+                foreach (var dbSubtask in task.Subtasks)
+                {
+                    subtasksInfo.Add(_taskInfoMapper.Map(dbSubtask, null, null));
+                }
+            }
+
+            TaskResponse response = _taskResponseMapper.Map(
+                task,
+                usersDataResponse.FirstOrDefault(x => x.Id == task.AuthorId),
+                usersDataResponse.FirstOrDefault(x => x.Id == task.AssignedUser.Id),
+                usersDataResponse.FirstOrDefault(x => x.Id == parentTaskAssignedTo),
+                department.Name,
+                usersDataResponse.FirstOrDefault(x => x.Id == task.ParentTask.AuthorId),
+                subtasksInfo);
+            response.Errors = errors;
+
+            return response;
+        }
+
+        private void Authorization(DbTask task, List<string> errors, out IGetDepartmentResponse department)
+        {
+            List<DbProjectUser> projectUsers = null;
+            if (task != null)
+            {
+                projectUsers = _userRepository.GetProjectUsers(task.ProjectId, false).ToList();
+            }
+
+            Guid requestUserId = _httpContext.GetUserId();
+            
+            department = GetDepartment(requestUserId, errors);
+
+            bool isAdmin = _accessValidator.IsAdmin(requestUserId);
+
+            bool isProjectParticipant = false;
+            if (projectUsers != null)
+            {
+                isProjectParticipant =  projectUsers.FirstOrDefault(x =>
+                    x.UserId == requestUserId) != null;
+            }
+
+            bool isDepartmentDirector = false;
+            if (department != null)
+            {
+                isDepartmentDirector = department.DirectorUserId == requestUserId;
+            }
+
+            if (!isAdmin && !isProjectParticipant && !isDepartmentDirector)
```

Comment:
Now you do all request and after it you decide have this user rights or not. But if this user admin, we can do not send request to company service and go to user repository, or if user if project participant you again can don't send request to company service